src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,147 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * table.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = getHash(key);
+        int position = getPosition(hash, table.length);
+        if (table[position] == null) {
+            table[position] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K, V> current = table[position];
+            if (current.key == key || current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            while (current.next != null) {
+                if (current.key == key) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            if (current.key == key || current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            current.next = new Node<>(hash, key, value, null);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
```

Comment:
Let's look at another implementation
```suggestion
        public V getValue(K key) {
        Node<K, V> current = table[getBucketIndex(key)];
        while (current != null) {
            if (Objects.equals(current.key, key)) {
                return current.value;
            }
            current = current.next;
        }
        return null;
    }
```